user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -2029,6 +2040,10 @@ define([
     return path.charAt(0) === REPOSITORY_ROOT_PATH;
   }
 
+  function isRootPath(path) {
+    return isRepositoryRootPath(path) || path === "pvfs://";
```

Comment:
This is a simplification of the concept of generic files and generic file paths.
In general, and even though this is not currently used in the PUC side, these could include non-pvfs URLs, such as `recents://`.

Checking directly if a path is a root path, would require checking for the structure `<scheme>://`. Ideally, we'd be using a shared library to hold this logic, but we don't currently have one we could use for plain JS. The closest are the methods in the `Encoder` module.

Another way would be to use the information on the `GenericFile` data structure returned by the endpoint. A root path is a file whose `parentPath` is `null`.

Also, either case, I think it would be better to call these "provider roots". Keeping this method, it would be called `isProviderRootPath(.)`.

Reply:
Closing this PR, I addressed this comment in https://github.com/pentaho/pentaho-platform/pull/5641